src/sentry/tagstore/snuba/backend.py

```diff
@@ -157,33 +161,61 @@ def __get_tag_keys_for_projects(
         limit=1000,
         keys=None,
         include_values_seen=True,
+        use_cache=False,
         **kwargs
     ):
-        filters = {"project_id": projects}
+        filters = {"project_id": sorted(projects)}
         if environments:
-            filters["environment"] = environments
+            filters["environment"] = sorted(environments)
         if group_id is not None:
             filters["group_id"] = [group_id]
         if keys is not None:
-            filters["tags_key"] = keys
+            filters["tags_key"] = sorted(keys)
         aggregations = [["count()", "", "count"]]
 
         if include_values_seen:
             aggregations.append(["uniq", "tags_value", "values_seen"])
         conditions = []
 
-        result = snuba.query(
-            start=start,
-            end=end,
-            groupby=["tags_key"],
-            conditions=conditions,
-            filter_keys=filters,
-            aggregations=aggregations,
-            limit=limit,
-            orderby="-count",
-            referrer="tagstore.__get_tag_keys",
-            **kwargs
-        )
+        should_cache = use_cache and group_id is None
+        result = None
+
+        if should_cache:
```

Comment:
I don't think you can fully remove start and end from the cache key or you will be returning the wrong results:
get_tag_keys_for_projects between now and 5 minutes ago will return the same result from cache as a call to the same method over three months since they will resolve to the same cache key.
What I meant this morning was that you need to either have two different policies to add jitter to the key (one when the dates are present since you quantize the key and the other when they are not present) or you set default start and end when they are not passed right at the top of this method and then use a cache policy similar to the one you implemented yesterday + jitter.
But you cannot ignore one case because there would be a cache key conflict between the case where the user provides the parameters and the case where the user does not.

I would advise you to set a value for start and end when they are not passed so you do not need to calculate the key differently in different cases and you achieve the same result.